### DISCUSSION TITLE
Add XCTexpectFailure

### DISCUSSION THREAD

**Discussion by ZevEisenberg** [pointfreeco/swift-issue-reporting#58] [Ideas]
TCA [already uses](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Sources/ComposableArchitecture/TestStore.swift#L2431) a dyanmic-overlay-esque version of [`XCTExpectFailure`](https://developer.apple.com/documentation/xctest/xctexpectedfailure) to show what tests _would have_ failed, had exhaustive testing been enabled. This is slick, and I'd like to do something similar in a testing library that already uses xctest-dynamic-overlay. I could borrow the code from TCA, but it seems like a natural fit for this library to include it.

**Comment by stephencelis** [comment]
@ZevEisenberg We'd definitely be down for that, but haven't prioritized. The main thing to recapture is that Apple's `XCTExpectFailure` is a continuation and passes the value returned from the closure to the caller. We can do this too, but it'll make the code path a little more precarious (it'll have to crash if the symbol isn't found).

We'll also want to make sure it's compiled out of Linux.

Are you interested in trying to PR this?

**Reply by ZevEisenberg** [reply]
>(it'll have to crash if the symbol isn't found).

Do you mean that it'll need to `fatalError` if the `dlsym` call returns `nil`?

Anyway, I'll put up a janky PR soon because it's easier to talk about bad code than theoretical code.

**Comment by ZevEisenberg** [comment]
PR is up: https://github.com/pointfreeco/xctest-dynamic-overlay/discussions/58
